lib/Folder/FolderManager.php

```diff
@@ -431,10 +431,26 @@ private function getCircles(int $id): array {
 		$circles = $this->getAllApplicable()[$id] ?? [];
 		$circles = array_map(fn (string $singleId): ?Circle => $this->getCircle($singleId), array_keys($circles));
 
+		// get nested teams
+		$nested = [];
+		foreach ($circles as $circle) {
+			try {
+				$inherited = $circle?->getInheritedMembers(true) ?? [];
+			} catch (\Exception $e) {
+				$this->logger->notice('could not get nested teams', ['exception' => $e]);
+				continue;
+			}
+			foreach ($inherited as $entry) {
+				if ($entry->getUserType() === Member::TYPE_CIRCLE) {
+					$nested[] = $entry->getBasedOn();
+				}
+			}
+		}
```

Comment:
Does this properly account for triple nested teams as well?

Reply:
Yes, inheritedmembers is using memberships table and will returns all nested memberships